alicloud/data_source_alicloud_db_instances.go

```diff
@@ -0,0 +1,282 @@
+package alicloud
+
+import (
+	"log"
+	"regexp"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/rds"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAlicloudDBInstances() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAlicloudDBInstancesRead,
+
+		Schema: map[string]*schema.Schema{
+			"name_regex": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateNameRegex,
+			},
+			"engine": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validateAllowedStringValue([]string{
+					string(MySQL),
+					string(SQLServer),
+					string(PPAS),
+					string(PostgreSQL),
+				}),
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Optional: true,
+				// please follow the link below to see more details on available statusesplease follow the link below to see more details on available statuses
+				// 实例状态表
```

Comment:
Please avoid to use Chinese annotation.
By the way, I think there is no need to list all of status values. In general, most uses want to get running and restoring instance. So, I suggest its validation can be removed.

Reply:
please find update